app/services/fastly_config/update.rb

```diff
@@ -0,0 +1,58 @@
+module FastlyConfig
+  # Handles updates to our Fastly configurations
+  class Update
+    FASTLY_CONFIGS = %w[Snippets].freeze
+
+    class << self
+      def run(configs: FASTLY_CONFIGS)
+        validate_configs(configs)
+
+        fastly = Fastly.new(api_key: ApplicationConfig["FASTLY_API_KEY"])
+        service = fastly.get_service(ApplicationConfig["FASTLY_SERVICE_ID"])
+        active_version = get_active_version(service)
+        config_handlers = configs.map { |config| "FastlyConfig::#{config}".constantize.new(fastly, active_version) }
+        configs_updated = config_handlers.any?(&:update_needed?)
+
+        return unless configs_updated
+
+        new_version = active_version.clone
+        config_handlers.each { |config_handler| config_handler.update(new_version) }
+        new_version.activate!
+        log_to_datadog(configs, new_version)
+        Rails.logger.info("Fastly updated to version #{new_version.number}.")
+      rescue Fastly::Error => e
+        error_msg = JSON.parse(e.message)
+        raise e unless unauthorized_error?(error_msg) && Rails.env.development?
+
+        nil
+      end
+
+      private
+
+      def get_active_version(service)
+        service.versions.reverse_each.detect(&:active?)
+      end
```

Comment:
I want to make note of this logic in case it becomes an issue in the future.

This logic is to find the active version on Fastly. Unfortunately, it's not on the service object itself. The other way to find it is to parse out the details on the service object, pluck the version number of the active version, and make another API call to get the active version object. I'm trying to keep API calls to a minimum.

Versions is a slowly growing infinite Array. Anytime we make an update to Fastly, we create a new version. As of this writing we're on version 125. It's common for the active version to be the most recent, but that's not always true. This solution should be performant enough for the foreseeable future as the Array grows. And there's no way to delete a version, you'd have to create an entirely new service to start the count over.